aptos-move/aptos-vm/src/validator_txns/dkg.rs

```diff
@@ -0,0 +1,126 @@
+// Copyright © Aptos Foundation
+
+use crate::{
+    aptos_vm::get_or_vm_startup_failure,
+    errors::expect_only_successful_execution,
+    move_vm_ext::{AptosMoveResolver, SessionId},
+    system_module_names::{FINISH_WITH_DKG_RESULT, RECONFIGURATION_WITH_DKG_MODULE},
+    validator_txns::dkg::{
+        ExecutionFailure::{Expected, Unexpected},
+        ExpectedFailure::*,
+    },
+    AptosVM,
+};
+use aptos_types::{
+    dkg::{DKGNode, DKGState, DKGTrait, DefaultDKG},
+    fee_statement::FeeStatement,
+    move_utils::as_move_value::AsMoveValue,
+    on_chain_config::OnChainConfig,
+    transaction::{ExecutionStatus, TransactionStatus},
+};
+use aptos_vm_logging::log_schema::AdapterLogSchema;
+use aptos_vm_types::output::VMOutput;
+use move_core_types::{
+    account_address::AccountAddress,
+    value::{serialize_values, MoveValue},
+    vm_status::{AbortLocation, StatusCode, VMStatus},
+};
+use move_vm_types::gas::UnmeteredGasMeter;
+
+enum ExpectedFailure {
+    MissingResourceDKGState = 0,
+    MissingResourceInprogressDKGSession,
+    EpochNotCurrent,
```

Comment:
Are these constants 0, etc. defined in Move as well?

Reply:
the Move part will probably land after this. Should be fine?